src/modules/sway/bar.cpp

```diff
@@ -21,15 +21,39 @@ BarIpcClient::BarIpcClient(waybar::Bar& bar) : bar_{bar} {
 
   signal_config_.connect(sigc::mem_fun(*this, &BarIpcClient::onConfigUpdate));
   signal_visible_.connect(sigc::mem_fun(*this, &BarIpcClient::onVisibilityUpdate));
+  signal_urgency_.connect(sigc::mem_fun(*this, &BarIpcClient::onUrgencyUpdate));
+  signal_last_modifier_.connect(sigc::mem_fun(*this, &BarIpcClient::onModifierPress));
+  signal_mode_.connect(sigc::mem_fun(*this, &BarIpcClient::onModeUpdate));
 
-  ipc_.subscribe(R"(["bar_state_update", "barconfig_update"])");
+  // Subscribe to non bar events to determine if the modifier key press is followed by another
+  // action.
+  ipc_.subscribe(R"(["bar_state_update", "barconfig_update", "workspace", "mode", "binding"])");
   ipc_.signal_event.connect(sigc::mem_fun(*this, &BarIpcClient::onIpcEvent));
+  ipc_.signal_cmd.connect(sigc::mem_fun(*this, &BarIpcClient::onCmd));
+  ipc_mode_.subscribe(R"(["mode"])");
+  ipc_mode_.signal_event.connect(sigc::mem_fun(*this, &BarIpcClient::onEventMode));
+  ipc_workspace_.subscribe(R"(["workspace"])");
+  ipc_workspace_.signal_event.connect(sigc::mem_fun(*this, &BarIpcClient::onEventUrgency));
```

Comment:
You don't need to create multiple ipc connections to handle different event types; one connection is already enough.
```c++
void BarIpcClient::onIpcEvent(const struct Ipc::ipc_response& res) {
  switch(res.type) {
  case IPC_EVENT_BARCONFIG_UPDATE:
     ...
     break;
  case IPC_EVENT_BAR_STATE_UPDATE:
     ...
     break;
  case IPC_EVENT_WORKSPACE:
     ...
     break;
  case IPC_EVENT_MODE:
     ...
  }
```

Reply:
Thanks for the tip!